filament/backend/test/test_FeedbackLoops.cpp

```diff
@@ -66,132 +120,128 @@ TEST_F(BackendTest, FeedbackLoops) {
 
         // Create a program.
-        ProgramHandle downsampleProgram;
+        ProgramHandle program;
         {
-            ShaderGenerator shaderGen(downsampleVs, downsampleFs, sBackend, sIsMobilePlatform);
+            ShaderGenerator shaderGen(fullscreenVs, fullscreenFs, sBackend, sIsMobilePlatform);
             Program prog = shaderGen.getProgram();
-            Program::Sampler psamplers[1];
-            psamplers[0].binding = 0;
-            psamplers[0].name = utils::CString("tex");
-            psamplers[0].strict = false;
+            Program::Sampler psamplers[] = { utils::CString("tex"), 0, false };
             prog.setSamplerGroup(0, psamplers, sizeof(psamplers) / sizeof(psamplers[0]));
-            downsampleProgram = getDriverApi().createProgram(std::move(prog));
+            prog.setUniformBlock(1, utils::CString("params"));
+            program = getDriverApi().createProgram(std::move(prog));
         }
 
         TrianglePrimitive triangle(getDriverApi());
 
-        auto defaultRenderTarget = getDriverApi().createDefaultRenderTarget(0);
-
-        // Create one texture with two miplevels.
+        // Create a texture.
         auto usage = TextureUsage::COLOR_ATTACHMENT | TextureUsage::SAMPLEABLE;
         Handle<HwTexture> texture = getDriverApi().createTexture(
-                    SamplerType::SAMPLER_2D,            // target
-                    2,                                  // levels
-                    TextureFormat::RGBA8,               // format
-                    1,                                  // samples
-                    512,                                // width
-                    512,                                // height
-                    1,                                  // depth
-                    usage);                             // usage
-
-        // Create a RenderTarget for miplevel 1.
-        Handle<HwRenderTarget> renderTarget = getDriverApi().createRenderTarget(
-                TargetBufferFlags::COLOR,
-                256,                                       // width of miplevel
-                256,                                       // height of miplevel
-                1,                                         // samples
-                { texture, 1, 0 },                         // color level = 1
-                {},                                        // depth
-                {});                                       // stencil
-
-        // Fill the texture with interesting colors.
-        const size_t size = 512 * 512 * 4;
+            SamplerType::SAMPLER_2D, kNumLevels, kTexFormat, 1, kTexWidth, kTexHeight, 1, usage);
+
+        // Create a RenderTarget for each miplevel.
+        Handle<HwRenderTarget> renderTargets[kNumLevels];
+        for (uint8_t level = 0; level < kNumLevels; level++) {
+            printf("Level %d: %d x %d\n", level, kTexWidth >> level, kTexHeight >> level);
```

Comment:
You should use our iostream/logger because it'll work nicer on Android if we ever run this test on Android.